cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -186,7 +189,13 @@ func (scaleSet *ScaleSet) getCurSize() (int64, error) {
 
 // GetScaleSetSize gets Scale Set size.
 func (scaleSet *ScaleSet) GetScaleSetSize() (int64, error) {
-	return scaleSet.getCurSize()
+	// First, get the size of the ScaleSet reported by API
```

Comment:
Why are we ever in the flow of "get me the size of a scaleset" before the cache has a record of the scaleset? Would it be possible to enforce more serial sanity here, so that we populate the cache for newly observed scalesets (either literally new scalelets, or e.g., the cluster-autoscaler runtime has restarted and so has to repopulate its cache) as part of the flow of retrieving details about it (e.g., its size).

I think we definitely want to continue having all GET flows derive data from the cache, just wondering if we can improve the behavior of "cache miss" flows so that we don't have these weird edge cases that may slow down eventual consistency.